Fix #18

DevStarlight opened the pull request:
Don't merge until a new commit is comming with more changes

CodingCarlos committed:
Add download page

Also enabled HTML5 routing mode

CodingCarlos committed:
Download file from API without corruption

DevStarlight committed:
initial structure download-video

CodingCarlos commented on js/app.js:
```diff
@@ -24,19 +24,19 @@
-			enabled: true,
-			requireBase: false
-		});
+//		$locationProvider.html5Mode({
```
why commenting html5 mode?

CodingCarlos commented on pages/video-download/video-download.service.js:
```diff
@@ -10,10 +10,10 @@ function videoDownloadService(api) {
-				console.error('Unable to download file.');
-			}
-		});
+//		api.download(api.url + '/video/' + videoId + '/original?code=' + code, function (data, status, headers) {
```
how do we download the video if we comment the service who does the magic? xDDD 

(I know, it's done to avoid excessive requests while testing, no worry, I'm fixing it now 😆)

CodingCarlos commented on pages/video-download/video-download.view.html:
```diff
@@ -1,11 +1,26 @@
-
-	<input type="text" ng-model="VideoDownload.code">
-	<a ng-click="VideoDownload.download()" target="_blank">Download</a>
+<!--YEEEEEPALE!! {{VideoDownload.id}}-->
```
Shall remove this line, to avoid keeping so much thrash in the code ;)

CodingCarlos commented on pages/video-download/video-download.view.html:
```diff
@@ -1,11 +1,26 @@
+<!--YEEEEEPALE!! {{VideoDownload.id}}-->
+<div ng-controller="VideoDownloadController as VideoDownload" class="download-page">
+	<article>
+		<video poster="../../img/video1.jpg" class="video-download-page"></video>
```
`poster="../../img/video1.jpg"` Shall better be the `{{VideoDownload.video.data.poster}}`, that is the actual video poster.

Also, just to make sure, have we talked with Juan about default posters? No worry so much now about it, is just to know what's the actual status ;)

DevStarlight committed:
added gallery state to allow video download screen going back to the gallery
changed css path since html5mode was not working properly & re-enable this piece of code
Added checked svg
Removed some useless code
Include 2 new mixins depending of portrait & landscape

DevStarlight committed:
added video properties so now it's able to reproduce in a good way

DevStarlight committed:
Merge branch 'dev' of github.com:Videona/platform-frontend into 154720332-download-video

# Conflicts:
#	dist/css/style.min.css
#	dist/js/app-debug.js
#	dist/js/app.min.js
#	dist/pages/register/register.view.html

DevStarlight committed:
removing an img that should not be here

DevStarlight commented on dist/pages/video-download/video-download.view.html:
```diff
@@ -0,0 +1,30 @@
+<div ng-controller="VideoDownloadController as VideoDownload" class="download-page">
+	<article>
+		<video controls ng-src="{{::VideoDownload.video.data.video}}" ng-attr-poster="{{::VideoDownload.video.data.video}}" preload="metadata" controlsList="nodownload" class="video-download-page"></video>
```
I've set the `preload="metadata"` to get the canvas its correct size... If having the player without size doesn't matter, then we could set this property to `none`

DevStarlight commented:
@CodingCarlos All your comments were fixed in my commits. There are 2 new issues from this one 
 (#22 & #23), so if none of these 2 are important, we can merge this one into dev =)

CodingCarlos commented:
With a fast pre-qa, I've seen that desktop layout is quite different than [marvel's desgin](https://marvelapp.com/7deb3f7/screen/37925879). I'm taking it, and I'm quite much sure it will be fixed before launch ;)

CodingCarlos commented:
There is so much chaos in color definitions. I'm feeling like color variables are not being used, and that's the main reason why we're using SASS. 

I'm currently working on standardizing colors and variables, and understand why do we have so much color variables if we're not using it, and we're putting handpicked colors.

So, these changes are being longer and bigger than expected.

CodingCarlos committed:
Desktop view fixes

DevStarlight commented:
It seems that everything is alright. This update makes the design much more accurate to the design. Thanks, @CodingCarlos!
